nftables getting-started guide

caseydavenport opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][DOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

Product Version(s):
<!--- Specify which versions of Calico, Calico Enterprise, and Calico Cloud your PR applies to. -->

Issue:
<!--- Add a link to the Jira ticket or GitHub issue, if applicable. --->

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

SME review:
- [ ] An SME has approved this change. 
<!--- SME approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

DOCS review:
- [ ] A member of the docs team has approved this change.
<!-- The Docs team must review and approve all changes. -->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

Merge checklist:
<!--- Mandatory for docs team members before merging code --->
- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed 


<!--- After you open your PR, ask for review from docs team:
  For community authors: tag @tigera/docs to ask for a review when your PR is ready --->

caseydavenport committed:
Add basic nftables dataplane getting started guide

caseydavenport committed:
Add to sidebar

ctauchen commented on sidebars-calico.js:
```diff
@@ -47,6 +47,7 @@ module.exports = {
             'getting-started/kubernetes/requirements',
             'getting-started/kubernetes/community-tested',
             'getting-started/kubernetes/quickstart',
+            'getting-started/kubernetes/nftables',
```
There's a larger conversation about how and where we document dataplane guides like this. But for now it makes most sense to group this further down next to the VPP guide.

ctauchen commented on calico/getting-started/kubernetes/nftables.mdx:
```diff
@@ -0,0 +1,193 @@
+---
+description: Install Calico using the tech-preview nftables data plane.
```
I can see this easily being missed when it goes GA. Can we scrap the tech preview language here?

ctauchen commented on calico/getting-started/kubernetes/nftables.mdx:
```diff
@@ -0,0 +1,193 @@
+---
+
+# Calico nftables data plane
+
```
```suggestion
:::note

This feature is tech preview. Tech preview features may be subject to significant changes before they become GA.

:::
```
We're using this boilerplate for page-level tech preview notices.

ctauchen commented on calico/getting-started/kubernetes/nftables.mdx:
```diff
@@ -0,0 +1,193 @@
+{{prodname}} nftables dataplane is currently in technical preview and may not be suitable for production use. Please report any issues you experience
+while using this feature.
+
+:::
```
Remove, in favor of boilerplate above.

ctauchen commented on calico/getting-started/kubernetes/nftables.mdx:
````diff
@@ -0,0 +1,193 @@
+
+1. Create a kubeadm configuration file that enables the nftables kube-proxy mode.
+
+   ```
````
```suggestion
   ```bash
```
Globally, for syntax highlighting.

ctauchen reviewed (commented):
A few suggestions for you. This is a good doc to get folks up and running with the new dataplane. It would be worth discussing how we ought to be presenting dataplane information generally in the docs. For both eBPF and VPP, we're relying on bespoke guides, not unlike this one, that were written to present the tech preview feature. But we should see how we can integrate these procedures as part of a standard installation procedure.

caseydavenport committed:
Update calico/getting-started/kubernetes/nftables.mdx

Co-authored-by: Christopher Tauchen <tauchen@gmail.com>

caseydavenport committed:
Code review comments

caseydavenport commented:
@ctauchen thanks for the review! I think I've addressed your comments.

> But we should see how we can integrate these procedures as part of a standard installation procedure.

Definitely agree, but I think it's larger than this PR.

It might be nice to have data plane selection more seamless integrated into the other install docs rather than tucked away in its own set. The only concern I see is complicating what would otherwise be a pretty straightforward and opinionated guide with information that isn't relevant to every user.

ctauchen commented:
> It might be nice to have data plane selection more seamless integrated into the other install docs rather than tucked away in its own set. The only concern I see is complicating what would otherwise be a pretty straightforward and opinionated guide with information that isn't relevant to every user.

Yes, I think it's all worth doing, but it will have to be part of a much larger effort. 

For now, do you want to shop this doc around a bit, or are you happy to merge? @caseydavenport

caseydavenport committed:
Fix some config

caseydavenport commented:
I think I'm good to merge this :+1:
